### PR TITLE
Big IsPresent$ clean up

### LINKED ISSUE #1706: Brain Maggot removed before trigger resolves doesn't reveal hand
I'm on the 1.6.54 snapshot and Brain Maggot was removed while its trigger was on the stack. The correct action should be that I see the opponent's hand [and technically choose a card, although it won't be exiled]. Instead, the ability resolved with no effect.

### PR DESCRIPTION
**Pull request by tool4ever**
Bonus fixes include more wrong use of CARDNAME and some others.

Closes #1706

### CODE REVIEW

File: forge-game/src/main/java/forge/game/cost/CostDiscard.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ else if (type.equals("DifferentNames")) {
             }
             if (sameName) {
                 for (Card c : handList) {
-                    if (CardLists.filter(handList, CardPredicates.nameEquals(c.getName())).size() > 1) {
+                    if (CardLists.count(handList, CardPredicates.nameEquals(c.getName())) > 1) {
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/cost/CostDiscard.java:177]: i think there is `Iterables.any` or something like that, that would stop if a valid card is found

Comment (tool4ever) [reply forge-game/src/main/java/forge/game/cost/CostDiscard.java:177]: Yea but we need > 1

**Review by Agetian — APPROVED** [review]

File: forge-gui/res/cardsfolder/a/auriok_steelshaper.txt [hunk 2]
```diff
@@ -3,8 +3,7 @@ ManaCost:1 W
 Types:Creature Human Soldier
 PT:1/1
 S:Mode$ ReduceCost | ValidCard$ Card | ValidSpell$ Activated.Equip | Activator$ You | Amount$ 1 | Description$ Equip costs you pay cost {1} less.
-S:Mode$ Continuous | Affected$ Creature.Soldier+YouCtrl,Creature.Knight+YouCtrl | AddPower$ 1 | AddToughness$ 1 | CheckSVar$ X | SVarCompare$ EQ1 | Description$ As long as CARDNAME is equipped, each creature you control that's a Soldier or a Knight gets +1/+1.
-SVar:X:Count$Valid Card.Self+equipped
+S:Mode$ Continuous | Affected$ Creature.Soldier+YouCtrl,Creature.Knight+YouCtrl | AddPower$ 1 | AddToughness$ 1 | IsPresent$ Card.Self+equipped | Description$ As long as CARDNAME is equipped, each creature you control that's a Soldier or a Knight gets +1/+1.
 SVar:EquipMe:Once
 AI:RemoveDeck:Random
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/a/auriok_steelshaper.txt:8]: May be better with `DeckNeeds:Type$Soldier|Knight` ??

File: forge-gui/res/cardsfolder/b/baru_fist_of_krosa.txt [hunk 3]
```diff
@@ -3,7 +3,7 @@ ManaCost:3 G G
 Types:Legendary Creature Human Druid
 PT:4/4
 T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Forest | TriggerZones$ Battlefield | Execute$ TrigPump | TriggerDescription$ Whenever a Forest enters the battlefield, green creatures you control get +1/+1 and gain trample until end of turn.
-A:AB$ Token | Cost$ Discard<1/Card.namedBaru, Fist of Krosa> | TokenAmount$ 1 | TokenScript$ g_x_x_wurm | TokenOwner$ You | TokenPower$ X | TokenToughness$ X | CostDesc$ Grandeur — Discard another card named CARDNAME: | SpellDescription$ Create an X/X green Wurm creature token, where X is the number of lands you control.
+A:AB$ Token | Cost$ Discard<1/Card.namedBaru, Fist of Krosa> | TokenAmount$ 1 | TokenScript$ g_x_x_wurm | TokenOwner$ You | TokenPower$ X | TokenToughness$ X | CostDesc$ Grandeur — Discard another card named Baru, Fist of Krosa: | SpellDescription$ Create an X/X green Wurm creature token, where X is the number of lands you control.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/b/baru_fist_of_krosa.txt:6]: Don't know when we'll get back to this again so might be worth adding
`DeckHints:Ability$Token & Type$Wurm`

File: forge-gui/res/cardsfolder/b/beasts_of_bogardan.txt [hunk 4]
```diff
@@ -3,6 +3,5 @@ ManaCost:4 R
 Types:Creature Beast
 PT:3/3
 K:Protection from red
-S:Mode$ Continuous | Affected$ Card.Self | AddPower$ 1 | AddToughness$ 1 | CheckSVar$ X | SVarCompare$ GE1 | Description$ CARDNAME gets +1/+1 as long as an opponent controls a nontoken white permanent.
-SVar:X:Count$Valid Permanent.White+OppCtrl+nonToken
+S:Mode$ Continuous | Affected$ Card.Self | AddPower$ 1 | AddToughness$ 1 | IsPresent$ Permanent.White+OppCtrl+nonToken | Description$ CARDNAME gets +1/+1 as long as an opponent controls a nontoken white permanent.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/b/beasts_of_bogardan.txt:6]: Maybe RemoveDeck:Random on this weirdo?

File: forge-gui/res/cardsfolder/b/blood_cursed_knight.txt [hunk 5]
```diff
@@ -2,7 +2,6 @@ Name:Blood-Cursed Knight
 ManaCost:1 W B
 Types:Creature Vampire Knight
 PT:3/2
-S:Mode$ Continuous | Affected$ Card.Self | AddPower$ 1 | AddToughness$ 1 | AddKeyword$ Lifelink | CheckSVar$ X | SVarCompare$ GE1 | Description$ As long as you control an enchantment, CARDNAME gets +1/+1 and has lifelink.
-SVar:X:Count$Valid Enchantment.YouCtrl
+S:Mode$ Continuous | Affected$ Card.Self | AddPower$ 1 | AddToughness$ 1 | AddKeyword$ Lifelink | IsPresent$ Enchantment.YouCtrl | Description$ As long as you control an enchantment, CARDNAME gets +1/+1 and has lifelink.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/b/blood_cursed_knight.txt:5]: DeckHas:Ability$LifeGain

File: forge-gui/res/cardsfolder/d/dire_wolves.txt [hunk 6]
```diff
@@ -2,7 +2,6 @@ Name:Dire Wolves
 ManaCost:2 G
 Types:Creature Wolf
 PT:2/2
-S:Mode$ Continuous | Affected$ Card.Self | AddKeyword$ Banding | CheckSVar$ CheckPLAINS | SVarCompare$ GE1 | Description$ CARDNAME has banding as long as you control a Plains.
-SVar:CheckPLAINS:Count$Valid Plains.YouCtrl
+S:Mode$ Continuous | Affected$ Card.Self | AddKeyword$ Banding | IsPresent$ Plains.YouCtrl | Description$ CARDNAME has banding as long as you control a Plains.
 AI:RemoveDeck:All
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/d/dire_wolves.txt:6]: Is this RemoveDeck:All because of Banding?
Maybe still add `SVar:BuffedBy:Plains` ?

**Pull request merged**